### DISCUSSION TITLE
語言世界的開源運動

### DISCUSSION THREAD

**Discussion by Xunhaoz** [1101-GS4719-C/Python-Community-GS4719-C#1332] [Ideas]
“抽象化”的視角下，水車、蒸汽機、發電機和GPT之間的差別在於抽象程度不同，或者，集成的知識-記憶密度不同。知識越抽象，能涵蓋的經驗對象就越廣，能解釋的原理就越普遍，所指導的工程技術對自然的改造力度就越大。

例如，水車的發明完全可以基於經驗觀察，用自然語言描述，被一個心靈手巧的匠人製作出來；而蒸汽機必須基於力學算法才能被改良。相比對水流的經驗觀察，流體力學的算法無疑是更為抽象的知識。此時，自然語言已經不夠精確了，需要藉助數學這樣的“形式語言”來表述。

形式語言是對自然語言的再抽象。它最大的特點是強調“規則”而非“意義”，因此非常適合描述“關係”與“變化”——1+1=2本身不表示任何意義，除非我們能帶入對象。形式語言的優勢在於一旦某種推理規則被確定，我們可以將萬物帶入進行計算，如此一來，語言便完成了“向通用性的跳轉”（見戴維·多伊奇《無窮的開始》）。

完成向通用性跳轉後，抽象活動便可以一直持續下去，這是很了不起的事。相比牛頓力學中的重力、質量這些相對直觀的概念，麥克斯韋必須先“發明”電荷、磁場、磁通量這些更為抽象的概念，並將使其形式語言化（數學化）才能指稱與描述更微觀層面的現象。

沒有形式語言系統（數理邏輯系統/編程語言）以及這種形式系統帶來的通用性，不可能有今天的科學技術。某種意義上，正是麥克斯韋用方程幫助我們“記住”了微觀世界的物理規律，開啟了“電時代”。

另一個重要節點是圖靈於1936年發表的那篇劃時代的論文——《論可計算數在判定問題中的應用》，天才圖靈通過一個思維模型論證了思維過程本身也是可以被抽象成一系列規則和算法的，“智時代”的一切成就都奠基於此。

一路抽像下來，在普通人眼裡，今天這個時代充滿了無數知識與技術黑箱，而這些黑箱之所以黑，是因為其中抽象概念和形式語言這樣的“黑話”實在太多了。

這裡我們要想起斯蒂格勒在破折號中指出的那個“矛盾”——知識的建構恰恰依賴於知識的外化。人類知識分工不斷使語言抽象化的同時，也製造了專業壁壘、就業機會和職業間的相對優勢，但抽象化的語言也拉低了我們認識世界的效率。在這個意義上，全人類生產力的提高是以限制個體發展潛力為代價的。

世界先是從曠野變成農田，今天成了一間巨大的辦公室，裡面有無數格子間，那些抽象術語和專業知識製造的專業壁壘就是格子間的隔板，隔行如隔山。

GPT是一把大錘，摧毀了所有抽象語言-黑話製造的隔板，把人類重新帶回自然語言的曠野上。

一切語言符號都彼此關聯，一個概念再复雜，只要你追問得夠多，它一定能被自然語言描述，只是由於黑話套黑話，路徑太過崎嶇，GPT天生有一種“逆溯抽象”的能力，可以點亮絕大部分黑話箱，讓我們藉助最平實的自然語言，就能從“語言世界”的任意一點出發，抵達另一點。

這裡引入了“語言世界”的概念，來自之前節目提到過的“三個世界”模型。我們看到聽到的現象世界是三個世界的疊加投影。

一個對象，它既是物理世界的一段波長，也是心理世界一類感受，在語言世界，我們被標記為“紅”，並關聯著無數其他符號，構成了關於紅色的知識與故事。這些知識與故事反過來塑造我們對紅的感受。語言符號最早是我們認識的產物，然後成為了我們認識世界的中介，最終成了世界的圖式與象徵，語言自成一個世界，不僅映射現實，也生成現實。

我們的現象世界就像一個“可視化界面”，支撐這個界面運行的，是人類數千年來累積下來的語言符號以及由這些語言符號構成的“世界知識”。在這個意義上，語言是人類世界的源代碼。

即使機器沒有人類的感官（無法感知物理世界）和情緒（無法體驗心理世界），僅靠遍歷語言世界——即便只是在概率上知道說完第一個字之後第二個字該說什麼——機器也能表現得“像個人”。

因為機器掌握了語言世界的結構，我們便獲得了一個界面，或一個API接口。通過這個界面與接口，語言世界和世界知識向我們敞開。換句話說，具象世界的抽象結構向我們敞開了，或者乾脆說——世界敞開了。

我願意把這種現象稱為“語言世界的開源運動”。

我們將技術定義為將事物“抽象化”並為其“編程”，眼下的線索已經很清晰了：人類先是用自然語言發明了形式語言，然後用形式語言與機器交互，編譯出了迄今為止最為精密的程序，現在居然讓機器反過來破譯了人類世界的源代碼——語言，今天的人類正在以一種前所未有的方式使用“語言”。

最近很多文科生很興奮，興致勃勃用自然語言來編程，一句話完成一幅插圖，兩句話寫出一個腳本，三句話做出一個程序，四句話生成一個遊戲，要知道這只是剛剛開始，未來我們能用一句話生成一切。程序員的那句Talk is cheap, show me the code 終於變成了 Code is cheap, show me the idea。

我們之所以能用自然語言來編程，是因為自然語言本身已經被反編程了。 （所謂反編譯，就是從低級語言反推高級語言的行為；從人類視角看，就是從機器碼反推出源代碼，如此一來，就可以重新修改和利用某個程序；但在機器視角，整個結構是翻轉的，在機器眼中，人類的自然語言就像我們眼中的二進制排列一樣毫無意義，對人類而言毫無意義的二進制排列對機器來說則像身體語言和自然語言一樣基本。翻轉到機器視角，反編譯是從自然語言反推出二進制機器碼。整個過程由形式語言中介完成。至此，“Hello world”與01排列開始相互映射，不是字符表層次的映射，而是“意義”層次的映射。）

在數學的意義上，人類發現了形式語言（數學語言-編程語言）與自然語言之間轉換函數。任何函數都可以被看成是一個黑箱，GPT就是那個黑箱，一個能解譯其他黑箱的黑箱。

人類通過極致的抽象來破譯了抽象，通過中介中介回到了事物自身。這就是“自然語言通用化”的意義。

以上是第一部分——描述一種變化。破題的關鍵是“通用性”，能源與語言可以被視為一種通用介質，其他事物必須都根據這些介質的特性，按照特定的規則才能組織起來。能源創造了物質世界的通用性，語言創造了精神世界的通用性，以數學、邏輯為代表的形式語言貫通了兩者，如今兩者匯流，將會對既有的生產與生活模式產生巨大衝擊，從前的積淤阻塞將被逐漸清除，新邊界的出現也將帶來新的規則與挑戰。

這絕非一次技術迭代那麼簡單，這是前所未有的革命性事件。

**Comment by boson13579** [comment]
你的觀點中描述了水車、蒸汽機、發電機和GPT（即ChatGPT）之間的差異，並著重於抽象程度和知識-記憶密度的不同。根據你的描述，這些差異在於它們使用的抽象程度和形式語言的不同。

你指出水車的發明可以基於經驗觀察，使用自然語言描述，由心靈手巧的匠人製作而成。相比之下，蒸汽機需要基於力學算法進行改良，而這些算法屬於更為抽象的知識。蒸汽機的改良需要使用形式語言（如數學）來描述，因為自然語言已經不夠精確。

**Comment by boson13579** [comment]
你還提到了形式語言的特點，它強調規則而非意義，適合描述關係和變化。形式語言的優勢在於一旦確定了某種推理規則，就可以將各種情況帶入進行計算，從而實現通用性。這種通用性使得抽象活動得以持續進行，並推動了科學技術的發展。

你提到麥克斯韋的方程式和圖靈的論文對於抽象化的重要性。麥克斯韋使用方程式幫助我們記住了微觀世界的物理規律，而圖靈的論文則將思維過程抽象成規則和算法，為智能時代的發展奠定了基礎。

**Comment by boson13579** [comment]
你提到了語言世界的概念，認為我們所看到和聽到的現象世界是三個世界的投影。語言世界是由語言符號和世界知識構成的，它不僅映射現實，還生成現實。

GPT在語言世界中的作用是通過處理自然語言文本來提供各種信息和交流。它可以理解和生成文本，並基於前面的內容進行推理和回答問題。GPT擁有龐大的知識庫，可以通過閱讀和處理大量的文本資料來學習和獲取知識。

**Comment by boson13579** [comment]
然而，與前面提到的水車、蒸汽機和發電機相比，GPT在抽象程度和知識-記憶密度方面存在一些不同。儘管GPT可以處理抽象的語言和概念，但它的知識庫是基於先前訓練的大量文本資料，而不是基於具體物理法則或數學模型。

GPT不像發電機一樣直接轉換能源，也不像蒸汽機一樣通過物理運動來產生功率。它是一種虛擬的智能系統，運行在計算機中，通過算法和模型進行信息處理和生成。GPT的功能主要集中在語言理解、文本生成和回答問題等方面。

**Comment by boson13579** [comment]
總之，水車、蒸汽機、發電機和GPT代表了不同時期和不同領域的技術發展。水車和蒸汽機是機械工程領域的重要發明，利用物理運動和力學原理來實現能源轉換和機械功率輸出。發電機則是電力工程領域的關鍵技術，將機械能轉換為電能。而GPT則是人工智能領域的一個代表，通過處理自然語言文本實現信息處理和生成。
